src/ansys/simai/core/data/model_configuration.py

```diff
@@ -0,0 +1,257 @@
+        if boundary_conditions is not None:
+            self.input.boundary_conditions = list(boundary_conditions.keys())
+        self.build_preset = build_preset
+        self.continuous = continuous
+        if fields is not None:
+            if fields.get("surface_input"):
+                self.input.surface = [fd.get("name") for fd in fields["surface_input"]]
+
+            if fields.get("surface"):
+                self.output.surface = [fd.get("name") for fd in fields["surface"]]
+
+            if fields.get("volume"):
+                self.output.volume = [fd.get("name") for fd in fields["volume"]]
+
+        self.simulation_volume = simulation_volume
+        if global_coefficients is not None:
+            gcs = [
+                GlobalCoefficientUnit(**gc) if isinstance(gc, dict) else gc
+                for gc in global_coefficients
+            ]
+            self.global_coefficients = gcs
+        if input is not None:
+            self.input = input
+        if output is not None:
+            self.output = output
+
+    def _to_payload(self):
+        """Constracts the payload for a build request."""
+
+        bcs = {}
+        if self.input.boundary_conditions is not None:
+            bcs = {bc_name: {} for bc_name in self.input.boundary_conditions}
+
+        sample_metadata = self.project.sample.fields.get("extracted_metadata")
+
+        surface_input_fld = []
+        if self.input.surface is not None:
+            surface_input_fld = [
+                fd
+                for fd in sample_metadata.get("surface").get("fields")
+                if fd.get("name") in self.input.surface
+            ]
+
+        surface_fld = []
+        if self.output.surface is not None:
+            surface_fld = [
+                fd
+                for fd in sample_metadata.get("surface").get("fields")
+                if fd.get("name") in self.output.surface
+            ]
+
+        volume_fld = []
+        if self.output.volume is not None:
+            volume_fld = [
+                fd
+                for fd in sample_metadata.get("volume").get("fields")
+                if fd.get("name") in self.output.volume
+            ]
+
+        gcs = []
+        if self.global_coefficients is not None:
+            gcs = [asdict(gc) for gc in self.global_coefficients]
+
+        flds = {
+            "surface": surface_fld,
+            "surface_input": surface_input_fld,
+            "volume": volume_fld,
+        }
+        return {
+            "boundary_conditions": bcs,
+            "build_preset": self.build_preset,
+            "continuous": self.continuous,
+            "fields": flds,
+            "global_coefficients": gcs,
+            "simulation_volume": self.simulation_volume,
+        }
+
+    def compute_global_coefficient(self):
+        """Computes the results of the formula for all Global Coefficients with respect to the project's sample."""
+
+        return [
+            self.project.compute_gc_formula(
```

Comment:
This should be parallelized

Reply:
I don't see an immediate reason to have them parallelized. It seems to me that parallelization at this point would be "good to have" than "should have". If this results in a bottleneck in future, we could improve it then.